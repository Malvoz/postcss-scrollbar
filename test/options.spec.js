import postcss from 'postcss';
import { stripIndent } from 'common-tags';
import plugin from '../src';

let from, to;

describe('options: ', () => {
  test('default options', () => {
    let input = stripIndent`
      .test {
        scrollbar-width: auto;
      }
    `;

    return postcss([plugin()])
      .process(input, { from, to })
      .then(result => {
        expect(result.css).toMatchSnapshot();
      });
  });

  test('with options', () => {
    let input = stripIndent`
      .test {
        scrollbar-width: none;
      }
    `;

    return postcss([plugin({ edgeAutohide: true })])
      .process(input, { from, to })
      .then(result => {
        expect(result.css).toMatchSnapshot();
      });
  });
});
