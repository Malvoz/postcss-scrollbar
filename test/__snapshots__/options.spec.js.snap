// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options:  default options 1`] = `
".test::-webkit-scrollbar {
  width: initial;
}
.test {
  -ms-overflow-style: auto;
  scrollbar-width: auto;
}"
`;

exports[`options:  with options 1`] = `
".test::-webkit-scrollbar {
  width: 0;
}
.test {
  -ms-overflow-style: -ms-autohiding-scrollbar;
  scrollbar-width: none;
}"
`;
